### PR TITLE
Add dumux.sh script to test dumux setup

### PR DESCRIPTION
**Pull request by samuelburbulla**
Related to https://github.com/adedner/dune-testpypi/pull/11

### CODE REVIEW

File: testing/dumux.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+# clone dumux
+echo "cloning dumux master"
+clonemodule dumux "$dumux_url" master master
+echo "done"
+
+# configure
+../repos/dune-common/bin/dunecontrol --module=dumux all
+
+pushd dumux/test/python
+python test_1p.py
```
Comment (timokoch) [inline testing/dumux.sh:23]:
```suggestion
python test_gridgeometry.py
python test_1p.py
```

File: testing/dumux.sh [hunk 2]
```diff
@@ -1,20 +1,21 @@
...
 dumux_url="https://git.iws.uni-stuttgart.de/dumux-repositories/dumux.git"
 
 # setup external venv and install the prepared packages needed
 # for dumux
 python3 -m venv dune-env
 source dune-env/bin/activate
-pip install mpi4py
-pip install --pre --find-links file://$PWD/../dist dune.common dune.grid dune.geometry dune.localfunctions dune.istl
+pip install mpi4py requests
+pip install --pre --find-links file://$PWD/../dist dune.common dune.grid dune.geometry dune.localfunctions dune.istl dune.alugrid
```
Comment (timokoch) [inline testing/dumux.sh]: Alugrid is not used in the Python tests and should definitely not be required for dumux (only optional). What was the error message here?

**Comment by timokoch** [comment]
I added a python unit test that doesn't need user parameters so we can see if that one works first.

**Comment by timokoch** [comment]
> I added a python unit test that doesn't need user parameters so we can see if that one works first.

Ok great the unit tests passes already. I'll try to fix the shared lib thing

File: testing/dumux.sh [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+. ../package
+python -m venv dune-env
+. dune-env/bin/activate
+pip install mpi4py requests
+
+export DUNE_CONTROL_PATH=".:../repos"
+
+# dumux Python bindings currently only work with shared libs
+export DUNE_CMAKE_FLAGS="-DBUILD_SHARED_LIBS=ON"
+
```
Comment (timokoch) [inline testing/dumux.sh]:
```suggestion
```

File: testing/dumux.sh [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+# clone dumux
+echo "cloning dumux master"
+clonemodule dumux "$dumux_url" master master
+echo "done"
+
+# configure
+../repos/dune-common/bin/dunecontrol --module=dumux all
```
Comment (timokoch) [inline testing/dumux.sh]:
```suggestion
DUMUX_CMAKE_FLAGS="-DBUILD_SHARED_LIBS=1" dunecontrol --opts=dumux/cmake-opts --only=dumux all
```

File: testing/dumux.sh [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+# dumux Python bindings currently only work with shared libs
+export CMAKE_FLAGS="-DBUILD_SHARED_LIBS=ON"
+
+# clone dumux
+echo "cloning dumux master"
+clonemodule dumux "$dumux_url" master master
+echo "done"
+
+# configure
+DUMUX_CMAKE_FLAGS="-DBUILD_SHARED_LIBS=1" dunecontrol --opts=dumux/cmake-opts --only=dumux all
```
Comment (timokoch) [inline testing/dumux.sh]:
```suggestion
DUMUX_CMAKE_FLAGS="-DBUILD_SHARED_LIBS=1" ../repos/dune-common/bin/dunecontrol --opts=dumux/cmake.opts --only=dumux all
```

File: testing/dumux.sh [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+# dumux Python bindings currently only work with shared libs
+export CMAKE_FLAGS="-DBUILD_SHARED_LIBS=ON"
+
+# clone dumux
+echo "cloning dumux master"
+clonemodule dumux "$dumux_url" master master
+echo "done"
+
+# configure
+DUMUX_CMAKE_FLAGS="-DBUILD_SHARED_LIBS=1" ../repos/dune-common/bin/dunecontrol --opts=dumux/cmake.opts --only=dumux all
```
Comment (timokoch) [inline testing/dumux.sh]:
```suggestion
DUMUX_CMAKE_FLAGS="-DBUILD_SHARED_LIBS=1" ../repos/dune-common/bin/dunecontrol --opts=dumux/cmake.opts --module=dumux all
```

File: testing/dumux.sh [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+. ../package
+python -m venv dune-env
+. dune-env/bin/activate
+pip install mpi4py requests
+
+export DUNE_CONTROL_PATH=".:../repos"
+
+# dumux Python bindings currently only work with shared libs
+export CMAKE_FLAGS="-DBUILD_SHARED_LIBS=ON"
+
```
Comment (timokoch) [inline testing/dumux.sh]:
```suggestion
```

File: testing/dumux.sh [hunk 8]
```diff
@@ -0,0 +1,25 @@
+# this tests dumux as external package which provides
+# a Python package outside the dune namespace
+
+dumux_url="https://git.iws.uni-stuttgart.de/dumux-repositories/dumux.git"
+
+# source the clonemodule function and make the core module visible
+. ../package
+python -m venv dune-env
+. dune-env/bin/activate
+pip install mpi4py requests
```
Comment (timokoch) [inline testing/dumux.sh:10]:
```suggestion
pip install mpi4py requests
pip install --pre --find-links file://$PWD/../dist dune.common dune.grid dune.geometry dune.localfunctions dune.istl dune.alugrid
```

File: testing/dumux.sh [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+
+export DUNE_CONTROL_PATH=".:../repos"
+
+# clone dumux
+echo "cloning dumux master"
+clonemodule dumux "$dumux_url" master master
+echo "done"
+
+# configure
+DUMUX_CMAKE_FLAGS="-DBUILD_SHARED_LIBS=1" ../repos/dune-common/bin/dunecontrol --opts=dumux/cmake.opts --module=dumux all
```
Comment (timokoch) [inline testing/dumux.sh]:
```suggestion
DUMUX_CMAKE_FLAGS="-DBUILD_SHARED_LIBS=1" dunecontrol --opts=dumux/cmake.opts --only=dumux all
```

File: testing/dumux.sh (hunk 2 above)
Comment (adedner) [reply testing/dumux.sh]: I didn't check the message. Perhaps best to remove it again and retest before we merge?

Comment (timokoch) [reply testing/dumux.sh]:
```suggestion
pip install --pre --find-links file://$PWD/../dist dune.common dune.grid dune.geometry dune.localfunctions dune.istl
```

**Comment by samuelburbulla** [comment]
Looks good to me now! Feel free to merge.
I just rebased because some other unrelated commits had crept in.

**Pull request merged**